src/lib/core/CHIPTLVReader.cpp

```diff
@@ -54,6 +54,14 @@ void TLVReader::Init(const uint8_t * data, size_t dataLen)
     ImplicitProfileId = kProfileIdNotSpecified;
 }
 
+CHIP_ERROR TLVReader::Init(const ByteSpan & span)
+{
+    VerifyOrReturnError(!span.empty(), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(CanCastTo<uint32_t>(span.size()), CHIP_ERROR_INVALID_ARGUMENT);
+    Init(span.data(), static_cast<uint32_t>(span.size()));
```

Comment:
Two-arg Init() now takes `size_t`, so no need for the cast bits.

Reply:
updated